api-server/server/routes/remoteApis.js

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env node
+'use strict'
+
+const LOGGER = require("../logger").logger
+const express = require("express")
+const openapiSampler = require('openapi-sampler');
+const refParser = require('json-schema-ref-parser');
+const { api, _, connection } = require("@varkes/app-connector")
+module.exports = {
+    router: router
+}
+
+function getAll(req, res) {
+    LOGGER.debug("Getting all APIs")
+    var err = assureConnected()
+    if (err) {
+        res.status(400).send({ error: err })
+    } else {
+        api.findAll().then((result) => {
+            res.status(200).send(result);
+        }, (err) => {
+            res.status(err.statusCode, err.message)
+        })
+    }
+}
+
+function get(req, res) {
+    LOGGER.debug("Get API %s", req.params.api)
+    var err = assureConnected()
+    if (err) {
+        res.status(400).send({ error: err })
+    } else {
+        api.findOne(req.params.api).then((result) => {
+
+            let body = result;
+            body.id = req.params.api //comply with the api spec
+            if (body.events && body.events.spec && Object.keys(body.events.spec).length !== 0) { //an empty events.spec {} causes bug

```

Comment:
please redesign that method, it is causing me headache. Just move the inner part into a dedicated function with a meaningful name

Reply:
done..removed the inner implementation to a function named dereferenceApi